Parking + Perception Integration. cleaned. ready to merge

AadarshHegde123 opened the pull request:
Integration of Perception's cone detection into parking team's parking space detection. We have also integrated this perception pipeline to work within the overall integration scheme outlined by the Planning team.

Video attached showcasing this integration's functionality. For reference, the ParkingSim component is the component that runs Parking team's overall state machine logic (ie. starting in Scanning mode, checking if we found a goal parking space and updating state to Parking). The RoutePlanningComponent is the component that will actually run planning logic depending on what state we are in.

The overall component pipeline is ConeDetector3D (Perception Team) -> ParkingSpotsDetector3D (Parking Team) -> ParkingSim (Parking Team) -> RoutePlanningComponent (Shared with other Teams)

The actual integrated planning logic is not included in this PR.

https://github.com/user-attachments/assets/5c82f0df-3944-480f-8aec-3e04351125bb

Commit:
cleaned. ready to merge

KenC1014 committed:
cleanup and move constants

KenC1014 commented:
Constants are now moved under utils/constants.py

leizhenyu-lzy committed:
delete useless functions

krishauser commented:
Perception PR is not yet merged pending review. Wait for their updates.

Commit:
integrating new perception changes

leizhenyu-lzy committed:
Update parking_utils.py

fix bug

AadarshHegde123 commented:
Perception has converted cones to be obstacles and updated their PR. This has been integrated and tested on this branch.

AadarshHegde123 commented:
Decided "middle ground" approach for calibration files is that calibration team will have a script that can convert their yaml files to the format perception and parking wants with cameras.yaml. With this approach parking and perception PRs can remain unchanged and are both functional.

Commit:
all new updates from perception added and tested

Commit:
Merge branch 'parking_perception_integration_merge' of https://github.com/krishauser/GEMstack into parking_perception_integration_merge

krishauser commented:
Still awaiting resolution of mission_plan.py and utils.py comments.

krishauser reviewed (changes requested):
Trying to submit file specific comments again…

Commit:
fixing component naming + moving MissionPlan to mission.py

KenC1014 committed:
1. read transform from yaml config 2. read vehcle geometry from config 3. remove multiple obstacle detection component

AadarshHegde123 commented on GEMstack/onboard/perception/utils/constants.py:
Addressed in commit 71ae7ef

AadarshHegde123 commented on GEMstack/state/mission_plan.py:
Addressed in commit 25cdc43

AadarshHegde123 commented on GEMstack/knowledge/defaults/computation_graph.yaml:
Addressed in commit 25cdc43

KenC1014 committed:
include newest changes from perception team

KenC1014 committed:
update variable name from agent to obstacle

Commit:
adding latest from perception

Commit:
fixing bad import

AadarshHegde123 commented:
Updated to include some changes perception made after feedback.

Commit:
fix for klampt sim bug

krishauser committed:
Merge branch 's2025' into parking_perception_integration_merge

krishauser commented on GEMstack/onboard/planning/route_planning_component.py:
Is this still being used? If not please take it out.

AadarshHegde123 replied:
This is still being used and will be further populated in the soon to be released parking planning PR.

krishauser commented on GEMstack/state/obstacle.py:
Please put the new attributes in Obstacle, not ObstacleState

AadarshHegde123 replied:
This file is not authored by Parking team, we copy it from what perception has on their PR branch to ensure consistency. We looked at the file more closely we didn't think having a separate ObstacleState class was necessary. 

I have talked with Tianyu and we have agreed to delete ObstacleState, and move anything needed to Obstacle. Please see commit 077ec81

AadarshHegde123 replied:
We are also up to date with Perception's PR, so this PR merged on its own will include all of perception's updates.

krishauser reviewed (changes requested):
Some continued minor changes

Commit:
fixing obstacle

Commit:
Merge branch 'parking_perception_integration_merge' of https://github.com/krishauser/GEMstack into parking_perception_integration_merge

Commit:
making sure everything works

Ren990420 committed:
Update cone_detection.py

changed a few variable names

Ren990420 committed:
Update __init__.py

Removed ObstacleState

Ren990420 committed:
Update __init__.py

removed obstacle state

Ren990420 committed:
Create cone_detection.yaml

This is the default cone_detection.yaml. Verticals should modify this file and integrate with their planning & other functions.

Commit:
update naming to match perception

Commit:
Merge branch 'parking_perception_integration_merge' of https://github.com/krishauser/GEMstack into parking_perception_integration_merge

krishauser reviewed (approved):
LGTM

krishauser committed:
Merge branch 's2025' into parking_perception_integration_merge
